addon/components/uni-tooltip.js

```diff
@@ -17,11 +21,41 @@ export default Component.extend({
 
   onClick() {},
 
-  click() {
+  onOutsideClick() {
     if (!this.get('isAlternative')) {
-      this.set('isActive', !this.get('isActive'));
+      this._hideTooltip();
     }
+  },
 
+  click() {
     this.get('onClick')();
+
+    if (!this.get('isAlternative')) {
```

Comment:
Use a guard clause and return instead of having this outer if

```js
if (this.get('isAlternative')) {
    return;
}

if (this.get('isActive')) {
    return this._hideTooltip();
}
```